src/main/java/dbot/command/Command.java

```diff
@@ -0,0 +1,150 @@
+package dbot.command;
+
+import dbot.ui.Ui;
+import dbot.exception.DBotException;
+import dbot.storage.Storage;
+import dbot.tasklist.TaskList;
+
+import javax.swing.*;
+
+/**
+ * Command is an abstract class that is to be extended to represent any valid user commands supplied to DBot.
+ */
+public abstract class Command {
+
+    /** A String containing the description that this Command's Task should have. */
+    protected String description;
+
+    /** A flag representing whether this Command's Task should be marked as done. */
+    private boolean isDone;
+
+    /** The index of the Task that this command targets in the TaskList. */
+    private int targetIndex = -1;
+
+    /**
+     * Initialize a command with its corresponding Task to be marked as not done.
+     */
+    public Command() {
+        isDone = false;
+    }
+
+    /**
+     * Initialize a command with its corresponding Task to be marked as not done and to have the supplied
+     * description String.
+     *
+     * @param description A String containing the description that the Task corresponding to this Command
+     *                    should have.
+     */
+    public Command(String description) {
+        this.description = description;
+        isDone = false;
+    }
+
+    /**
+     * Initializes a Command with the specified target index for a 0-indexed TaskList.
+     *
+     * @param targetIndex An integer representing the target index of a 0-indexed TaskList.
+     */
+    public Command(int targetIndex) {
+        setTargetIndex(targetIndex);
+    }
+
+    /**
+     * Sets the target index instance variable.
+     *
+     * @param targetIndex A integer specifying the new target index for the 0-indexed TaskList.
+     */
+    protected void setTargetIndex(int targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    /**
+     * Returns the target index for the 0-indexed TaskList.
+     *
+     * @return An integer specifying the target index for the 0-indexed TaskList.
+     */
+    protected int getTargetIndex() {
+        return targetIndex;
+    }
+
+    /**
+     * Returns the description that was supplied to this Command.
+     *
+     * @return A String containing the description that was supplied to this Command.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Returns true if this Command is an Exit Command and false otherwise.
+     *
+     * @return True if this is an Exit Command and false otherwise.

```

Comment:
Hi, thank you for the review.
This method was defined to return false in the Command class so that all classes that inherit it would also return false unless they override it, which is done only in the ExitCommand class.

I do understand the confusion that this method can cause and recognise that a more elegant solution might be to use a static method in the ExitCommand class to check if the user input is equivalent to an exit command.

(An example implementation that gave me the above idea can be found at: https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/commands/ExitCommand.java)